### PR TITLE
Add fail fast on build failures of TestPlatform.sln

### PR DESCRIPTION
**Pull request by smadala**
- Build should fail if there is error in build or publish steps.
- Build wasn't failing with latest dotnet-cli due to this, But it was failing in "test" step.

### CODE REVIEW

File: scripts/build.ps1 [hunk 1]
```diff
@@ -190,6 +190,10 @@ function Invoke-Build
     & $dotnetExe build $TPB_Solution --configuration $TPB_Configuration -v:minimal -p:Version=$TPB_Version -p:CIBuild=$TPB_CIBuild -p:LocalizedBuild=$TPB_LocalizedBuild
     Write-Log ".. .. Build: Complete."
 
+    if ($lastExitCode -ne 0) {
```
Comment (harshjain2) [inline scripts/build.ps1]: DRY violation. A single function can be used.

Comment (harshjain2) [inline scripts/build.ps1]: Can we move the logic to check to $lastExitCode to Set-ScriptFailed function?

**Review by harshjain2 — APPROVED** [review]

File: scripts/build.ps1 (hunk 1 above)
Comment (smadala) [reply scripts/build.ps1]: Done.

File: scripts/build.ps1 [hunk 2]
```diff
@@ -604,8 +602,12 @@ function Get-ElapsedTime([System.Diagnostics.Stopwatch] $timer)
     return $timer.Elapsed
 }
 
-function Set-ScriptFailed
+function Set-ScriptFailed-OnError
```
Comment (codito) [inline scripts/build.ps1]: nit: function name doesn't match the Powershell naming convention.

Comment (smadala) [reply scripts/build.ps1]: Renamed.

**Review by codito — APPROVED** [review]

**Pull request merged**